Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java

```diff
@@ -52,6 +63,73 @@
     // Explicit default constructor declaration
     public VungleMediationAdapter(final AppLovinSdk sdk) { super( sdk ); }
 
+    @Override
+    public void loadNativeAd(MaxAdapterResponseParameters maxAdapterResponseParameters, Activity activity, MaxNativeAdAdapterListener maxNativeAdAdapterListener) {
+
+        //assume as is loaded
+        final Bundle serverParameters = maxAdapterResponseParameters.getServerParameters();
+        final boolean isNativeBanner = serverParameters.getBoolean( "is_native_banner");
+        final String placementId = maxAdapterResponseParameters.getThirdPartyAdPlacementId();
+
+        NativeAd nativeAd = new NativeAd(activity, placementId);
+        MediaView mediaView = new MediaView(activity);
+        ImageView iconView = new ImageView(activity);
+        AdConfig adConfig = new AdConfig();
+        NativeAdLayout nativeAdLayout = new NativeAdLayout(activity);
+        nativeAd.loadAd(adConfig, new NativeAdListener() {
+            @Override
+            public void onNativeAdLoaded(NativeAd nativeAd) {
+                MaxNativeAd maxNativeAd = new MaxNativeAd.Builder()
+                        .setAdFormat(MaxAdFormat.NATIVE)
+                        .setTitle(nativeAd.getAdTitle())
+                        .setBody(nativeAd.getAdBodyText())
+                        .setMediaView(mediaView)
+                        .setIconView(iconView)
+                        .build();
+
+                List<View> clickableViews = new ArrayList<>();
+                clickableViews.add(maxNativeAd.getIconView());
+                clickableViews.add(maxNativeAd.getMediaView());
+                nativeAd.registerViewForInteraction(nativeAdLayout,
```

Comment:
Should destroy this `nativeAd` instance after the display to trigger reporting.

Reply:
@HuFangshuai agreed updating